### ISSUE TITLE
Cannot process "asset mint and transfer" example when using tendermint consensus

### ISSUE THREAD

**Issue by majecty** [CodeChain-io/codechain#442]
Asset mint transaction was processed well.
But asset transfer transaction was not processed.

Below is the log.
```
#678404526 2018-07-19 18:45:55  IO Worker #0 INFO sync  New peer detected #Local V4:3487
#678404526 2018-07-19 18:45:55  IO Worker #0 INFO sync  New peer detected #Local V4:3486
#678404526 2018-07-19 18:46:05  IO Worker #3 INFO sync  New peer detected #Local V4:3488
#678404526 2018-07-19 18:46:23  http.worker0 INFO tx  Tx(0xb4b8…81dc) is applied
#678404526 2018-07-19 18:46:24  IO Worker #2 INFO tx  Tx(0xb4b8…81dc) is applied
#678404526 2018-07-19 18:46:24  Verifier #0 INFO tx  Tx(0xb4b8…81dc) is applied
#678404526 2018-07-19 18:46:34  IO Worker #0 INFO own_parcel  Parcel mined (hash 0x6929b36953443b3c54894b3a71d0f7adab516a2f70dcf8b148df50858d7db49e)
#678404526 2018-07-19 18:46:34  IO Worker #0 INFO miner  Submitted block imported OK. #1: 0xc9f4…3f6b
#678404526 2018-07-19 18:46:34  http.worker0 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:46:44  IO Worker #0 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:46:46  Verifier #3 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:47:06  IO Worker #0 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:47:06  Verifier #0 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:47:17  Verifier #3 WARN client  Stage 4 block verification failed for #2 (0x3704…00cb)
Error: Engine(NotAuthorized(0x129801a0dbf53d21f14ede99424165001758a1d3))
#678404526 2018-07-19 18:47:26  IO Worker #1 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:47:37  IO Worker #0 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:47:47  IO Worker #0 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:47:47  Verifier #1 WARN client  Stage 4 block verification failed for #2 (0x47a9…8d2e)
Error: Engine(NotProposer(Mismatch { expected: 0x84137e7a75043bed32e4458a45da7549a8169b4d, found: 0x9f194cba03ecc7004646414e795957ceca32cd20 }))
#678404526 2018-07-19 18:48:07  IO Worker #3 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:48:07  Verifier #2 WARN client  Stage 4 block verification failed for #2 (0xe4fd…2b0b)
Error: Engine(NotProposer(Mismatch { expected: 0x9f194cba03ecc7004646414e795957ceca32cd20, found: 0xcfd28c1279645febb7cf8802993291ef433a21f3 }))
#678404526 2018-07-19 18:48:27  IO Worker #3 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:48:47  IO Worker #1 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:49:07  IO Worker #3 INFO tx  Tx(0xe4ff…512d) is applied
#678404526 2018-07-19 18:49:27  IO Worker #3 INFO tx  Tx(0xe4ff…512d) is applied

```

**Comment by kseo** [comment]
@majecty Would you test it again? This might have been fixed by 4855a6ec2b82f1e008ecee90e2c633032a8708c1.

**Comment by majecty** [comment]
@kseo  Now, it shows a different error message

```
#254228631 2018-07-25 14:34:58  IO Worker #2 TRACE engine  Prevote timeout without enough votes.
#254228631 2018-07-25 14:34:58  IO Worker #2 TRACE engine  0x6fbd…b669 could not sign the message Invalid password
#254228631 2018-07-25 14:35:08  IO Worker #1 TRACE engine  Prevote timeout without enough votes.
#254228631 2018-07-25 14:35:08  IO Worker #1 TRACE engine  0x6fbd…b669 could not sign the message Invalid password
#254228631 2018-07-25 14:35:18  IO Worker #0 TRACE engine  Prevote timeout without enough votes.
#254228631 2018-07-25 14:35:18  IO Worker #0 TRACE engine  0x6fbd…b669 could not sign the message Invalid password
#254228631 2018-07-25 14:35:28  IO Worker #0 TRACE engine  Prevote timeout without enough votes.
#254228631 2018-07-25 14:35:28  IO Worker #0 TRACE engine  0x6fbd…b669 could not sign the message Invalid password
```

**Comment by majecty** [comment]
## node0
```sh
codechain@jmyang-desktop:~/codechain$ cargo run --  --db-path db/db0 --port 3485 --jsonrpc-port 8080 --engine-signer 6fbd7e04852ceedeab4dc2a3318cc8aa6922b669  --keys-path keys/key0 -c tendermint --password-path  password.txt
    Finished dev [unoptimized + debuginfo] target(s) in 0.18s
     Running `target/debug/codechain --db-path db/db0 --port 3485 --jsonrpc-port 8080 --engine-signer 6fbd7e04852ceedeab4dc2a3318cc8aa6922b669 --keys-path keys/key0 -c tendermint --password-path password.txt`
#51933094 2018-07-25 17:11:54  main INFO codechain  Starting client
#51933094 2018-07-25 17:11:54  main INFO codechain  Handshake Listening on 0.0.0.0:3485
#51933094 2018-07-25 17:11:54  main INFO discovery  Node runs with unstructured discovery
#51933094 2018-07-25 17:11:54  IO Worker #1 INFO sync  Sync extension initialized
#51933094 2018-07-25 17:11:54  main INFO codechain::rpc  RPC Listening on 0.0.0.0:8080
#51933094 2018-07-25 17:11:54  main INFO codechain::rpc  IPC Listening on /tmp/jsonrpc.ipc
#51933094 2018-07-25 17:11:54  main INFO test_script  Initialization complete
#51933094 2018-07-25 17:12:04  IO Worker #0 INFO sync  New peer detected #Local V4:3486
#51933094 2018-07-25 17:12:04  IO Worker #0 INFO sync  New peer detected #Local V4:3487
#51933094 2018-07-25 17:12:04  IO Worker #0 INFO sync  New peer detected #Local V4:3488
#51933094 2018-07-25 17:12:15  http.worker0 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:12:15  http.worker0 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:12:21  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:12:21  IO Worker #2 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:12:21  Verifier #2 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:12:21  Verifier #2 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:12:41  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:12:41  IO Worker #2 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:13:01  IO Worker #3 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:13:01  IO Worker #3 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:13:01  Verifier #0 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:13:01  Verifier #0 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:13:21  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:13:21  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:13:21  Verifier #1 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:13:21  Verifier #1 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:13:32  Verifier #3 WARN client  Stage 4 block verification failed for #1 (0x0541…445c)
Error: Engine(NotAuthorized(0x5cff4e788f4774b65764d450491be4bc9bc02e1c))
#51933094 2018-07-25 17:13:41  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:13:41  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:13:52  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:13:52  IO Worker #2 INFO tx  Tx(0x2527…9a02) is applied
#51933094 2018-07-25 17:13:52  Verifier #3 WARN client  Stage 4 block verification failed for #1 (0x4d5d…0844)
Error: Engine(NotProposer(Mismatch { expected: 0x354835363d1a0dae494c6dc619271e3685da37ad, found: 0xd3655cbdeea382702cd6261fc6697a96b01ca2ca }))
#51933094 2018-07-25 17:14:02  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#51933094 2018-07-25 17:14:02  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
```

## node1
```sh
   Running `target/debug/codechain --db-path db/db1 --port 3486 --jsonrpc-port 8082 --engine-signer e22ad7868b23c6d446347d49b8ef22210af1d7e5 --keys-path keys/key1 -c tendermint --bootstrap-addresses '127.0.0.1:3485' --password-path password.txt`
#129524503 2018-07-25 17:12:02  main INFO codechain  Starting client
#129524503 2018-07-25 17:12:02  main INFO codechain  Handshake Listening on 0.0.0.0:3486
#129524503 2018-07-25 17:12:02  main INFO discovery  Node runs with unstructured discovery
#129524503 2018-07-25 17:12:02  IO Worker #1 INFO sync  Sync extension initialized
#129524503 2018-07-25 17:12:02  main INFO codechain::rpc  RPC Listening on 0.0.0.0:8082
#129524503 2018-07-25 17:12:02  event.loop WARN jsonrpc_ipc_server::server  Removed existing file '/tmp/jsonrpc.ipc'.
#129524503 2018-07-25 17:12:02  main INFO codechain::rpc  IPC Listening on /tmp/jsonrpc.ipc
#129524503 2018-07-25 17:12:02  main INFO test_script  Initialization complete
#129524503 2018-07-25 17:12:04  IO Worker #2 INFO sync  New peer detected #Local V4:3485
#129524503 2018-07-25 17:12:22  IO Worker #1 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:12:22  IO Worker #1 INFO tx  Tx(0x2527…9a02) is applied
#129524503 2018-07-25 17:12:41  IO Worker #1 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:12:41  IO Worker #1 INFO tx  Tx(0x2527…9a02) is applied
#129524503 2018-07-25 17:13:01  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:13:01  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
#129524503 2018-07-25 17:13:12  IO Worker #0 INFO sync  New peer detected #Local V4:3488
#129524503 2018-07-25 17:13:12  IO Worker #0 INFO sync  New peer detected #Local V4:3487
#129524503 2018-07-25 17:13:21  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:13:21  IO Worker #2 INFO tx  Tx(0x2527…9a02) is applied
#129524503 2018-07-25 17:13:21  Verifier #0 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:13:21  Verifier #0 INFO tx  Tx(0x2527…9a02) is applied
#129524503 2018-07-25 17:13:32  Verifier #2 WARN client  Stage 4 block verification failed for #1 (0x0541…445c)
Error: Engine(NotAuthorized(0x5cff4e788f4774b65764d450491be4bc9bc02e1c))
#129524503 2018-07-25 17:13:41  IO Worker #3 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:13:41  IO Worker #3 INFO tx  Tx(0x2527…9a02) is applied
#129524503 2018-07-25 17:13:52  IO Worker #1 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:13:52  IO Worker #1 INFO tx  Tx(0x2527…9a02) is applied
#129524503 2018-07-25 17:13:52  Verifier #1 WARN client  Stage 4 block verification failed for #1 (0x4d5d…0844)
Error: Engine(NotProposer(Mismatch { expected: 0x354835363d1a0dae494c6dc619271e3685da37ad, found: 0xd3655cbdeea382702cd6261fc6697a96b01ca2ca }))
#129524503 2018-07-25 17:14:02  IO Worker #1 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:14:02  IO Worker #1 INFO tx  Tx(0x2527…9a02) is applied
#129524503 2018-07-25 17:14:02  Verifier #2 WARN client  Stage 4 block verification failed for #1 (0x5ea3…0692)
Error: Engine(NotProposer(Mismatch { expected: 0xd3655cbdeea382702cd6261fc6697a96b01ca2ca, found: 0x6fbd7e04852ceedeab4dc2a3318cc8aa6922b669 }))
#129524503 2018-07-25 17:14:22  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
#129524503 2018-07-25 17:14:22  IO Worker #2 INFO tx  Tx(0x2527…9a02) is applied
```

## node2
```sh
     Running `target/debug/codechain --db-path db/db2 --port 3487 --jsonrpc-port 8084 --engine-signer 354835363d1a0dae494c6dc619271e3685da37ad --keys-path keys/key2 -c tendermint --bootstrap-addresses '127.0.0.1:3485' --password-path password.txt`
#927858482 2018-07-25 17:11:57  main INFO codechain  Starting client
#927858482 2018-07-25 17:11:58  main INFO codechain  Handshake Listening on 0.0.0.0:3487
#927858482 2018-07-25 17:11:58  main INFO discovery  Node runs with unstructured discovery
#927858482 2018-07-25 17:11:58  IO Worker #0 INFO sync  Sync extension initialized
#927858482 2018-07-25 17:11:58  main INFO codechain::rpc  RPC Listening on 0.0.0.0:8084
#927858482 2018-07-25 17:11:58  event.loop WARN jsonrpc_ipc_server::server  Removed existing file '/tmp/jsonrpc.ipc'.
#927858482 2018-07-25 17:11:58  main INFO codechain::rpc  IPC Listening on /tmp/jsonrpc.ipc
#927858482 2018-07-25 17:11:58  main INFO test_script  Initialization complete
#927858482 2018-07-25 17:12:04  IO Worker #1 INFO sync  New peer detected #Local V4:3485
#927858482 2018-07-25 17:12:21  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
#927858482 2018-07-25 17:12:21  IO Worker #2 INFO tx  Tx(0x2527…9a02) is applied
#927858482 2018-07-25 17:12:41  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
#927858482 2018-07-25 17:12:41  IO Worker #2 INFO tx  Tx(0x2527…9a02) is applied
#927858482 2018-07-25 17:13:01  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
#927858482 2018-07-25 17:13:01  IO Worker #2 INFO tx  Tx(0x2527…9a02) is applied
#927858482 2018-07-25 17:13:10  IO Worker #1 INFO sync  New peer detected #Local V4:3488
#927858482 2018-07-25 17:13:12  IO Worker #0 INFO sync  New peer detected #Local V4:3486
#927858482 2018-07-25 17:13:21  IO Worker #1 INFO tx  Tx(0x08ff…79ea) is applied
#927858482 2018-07-25 17:13:21  IO Worker #1 INFO tx  Tx(0x2527…9a02) is applied
#927858482 2018-07-25 17:13:31  IO Worker #0 INFO miner  Submitted block imported OK. #1: 0x0541…445c
#927858482 2018-07-25 17:13:52  Verifier #2 WARN client  Stage 3 block verification failed for #1 (0x4d5d…0844)
Error: State(Parcel(TransactionAlreadyImported))
#927858482 2018-07-25 17:14:02  Verifier #0 WARN client  Stage 3 block verification failed for #1 (0x5ea3…0692)
Error: State(Parcel(TransactionAlreadyImported))
#927858482 2018-07-25 17:14:22  Verifier #3 WARN client  Stage 3 block verification failed for #1 (0xf66b…02db)
Error: State(Parcel(TransactionAlreadyImported))
```

## node3
```sh
unning `target/debug/codechain --db-path db/db3 --port 3488 --jsonrpc-port 8088 --engine-signer d3655cbdeea382702cd6261fc6697a96b01ca2ca --keys-path keys/key3 -c tendermint --bootstrap-addresses '127.0.0.1:3485'
--password-path password.txt`
#615911041 2018-07-25 17:12:00  main INFO codechain  Starting client
#615911041 2018-07-25 17:12:00  main INFO codechain  Handshake Listening on 0.0.0.0:3488
#615911041 2018-07-25 17:12:00  main INFO discovery  Node runs with unstructured discovery
#615911041 2018-07-25 17:12:00  IO Worker #2 INFO sync  Sync extension initialized
#615911041 2018-07-25 17:12:00  main INFO codechain::rpc  RPC Listening on 0.0.0.0:8088
#615911041 2018-07-25 17:12:00  event.loop WARN jsonrpc_ipc_server::server  Removed existing file '/tmp/jsonrpc.ipc'.
#615911041 2018-07-25 17:12:00  main INFO codechain::rpc  IPC Listening on /tmp/jsonrpc.ipc
#615911041 2018-07-25 17:12:00  main INFO test_script  Initialization complete
#615911041 2018-07-25 17:12:04  IO Worker #1 INFO sync  New peer detected #Local V4:3485
#615911041 2018-07-25 17:12:21  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:12:21  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:12:41  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:12:41  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:13:01  IO Worker #1 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:13:01  IO Worker #1 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:13:10  IO Worker #1 INFO sync  New peer detected #Local V4:3487
#615911041 2018-07-25 17:13:12  IO Worker #0 INFO sync  New peer detected #Local V4:3486
#615911041 2018-07-25 17:13:21  IO Worker #1 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:13:21  IO Worker #1 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:13:21  Verifier #0 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:13:21  Verifier #0 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:13:32  Verifier #3 WARN client  Stage 4 block verification failed for #1 (0x0541…445c)
Error: Engine(NotAuthorized(0x5cff4e788f4774b65764d450491be4bc9bc02e1c))
#615911041 2018-07-25 17:13:41  IO Worker #3 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:13:41  IO Worker #3 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:13:52  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:13:52  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:14:02  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:14:02  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:14:02  Verifier #1 WARN client  Stage 4 block verification failed for #1 (0x5ea3…0692)
Error: Engine(NotProposer(Mismatch { expected: 0xd3655cbdeea382702cd6261fc6697a96b01ca2ca, found: 0x6fbd7e04852ceedeab4dc2a3318cc8aa6922b669 }))
#615911041 2018-07-25 17:14:22  IO Worker #0 INFO tx  Tx(0x08ff…79ea) is applied
#615911041 2018-07-25 17:14:22  IO Worker #0 INFO tx  Tx(0x2527…9a02) is applied
#615911041 2018-07-25 17:14:22  Verifier #3 WARN client  Stage 4 block verification failed for #1 (0xf66b…02db)
Error: Engine(NotProposer(Mismatch { expected: 0x6fbd7e04852ceedeab4dc2a3318cc8aa6922b669, found: 0xe22ad7868b23c6d446347d49b8ef22210af1d7e5 }))
#615911041 2018-07-25 17:14:42  IO Worker #2 INFO tx  Tx(0x08ff…79ea) is applied
```

**Comment by kseo** [comment]
Fixed by a178660f00e214c29b9f73c40f7589bcd28f38a8, bf7822425c943d8410483b61854c45bd15e4fe39 and 0ddda0dc41cf8711f51c0f478f89f0e94f2ac7cb
